File: presentation/src/main/java/com/mbmc/template/ui/DataWrapper.java
package com.mbmc.template.ui;

public class DataWrapper<T> {

    public T data;
    public Throwable throwable;

    public DataWrapper(Throwable throwable) {
        this.throwable = throwable;
    }

    public DataWrapper(T data) {
        this.data = data;
    }

}

File: presentation/src/main/kotlin/com/mbmc/template/ui/DataWrapper.kt
package com.mbmc.template.ui

class DataWrapper<T> {
    var data: T? = null
    var throwable: Throwable? = null

    constructor(data: T) {
        this.data = data
    }

    constructor(throwable: Throwable) {
        this.throwable = throwable
    }
}